### ISSUE TITLE
Package Check Time

### ISSUE THREAD

**Issue by gabrielodom** [TransBioInfoLab/coMethDMR#2]
Per the Bioconductor build report, our package check timed out at 15 minutes: http://bioconductor.org/spb_reports/coMethDMR_buildreport_20210422091842.html

I am going through the vignettes now to see if we can make them faster (smaller data examples perhaps)?

**Comment by gabrielodom** [comment]
Vignette 1 takes 55 seconds to knit.
The `AnnotateResults()` function takes much longer than I expected.

**Comment by gabrielodom** [comment]
Vignette 2 takes 7 minutes 50 seconds to knit. We need to find a smaller subset of regions to analyze.

**Comment by gabrielodom** [comment]
Vignette 2 takes 5 minutes and 10 seconds to knit on Mac, so it's not just a Windows problem.

**Comment by gabrielodom** [comment]
I cut the data in two ways:
1. After we select regions from Chromosome 22, I retain (as an example only) the first 10 regions, then
2. We remove CpGs from the betas data frame which are not members of the first 10 regions on Chromosome 22

Now, Vignette 2 takes 2 minutes and 15 seconds to knit on Mac.

**Comment by gabrielodom** [comment]
The R CMD check takes 16 minutes 47 seconds (R marked duration: 14'40").
The most expensive components are:

- creating vignettes (1m 59.7s)
- checking whether package ‘coMethDMR’ can be installed (39.7s)
- checking foreign function calls (2m 5.2s)
- checking R code for possible problems (37.7s)
- checking for code/documentation mismatches (43.4s)
- checking examples (1m 57.8s)
    + NameRegion: 23.6s
    + CloseBySingleRegion: 22.3s
    + OrderCpGsByLocation: 20.8s
    + AnnotateResults: 8.2s
- checking examples with --run-donttest (1m 56.8s)
- Running ‘testthat.R’ [40s/46s] (45.6s)
- checking for unstated dependencies in vignettes (45.8s)
- checking re-building of vignette outputs (3m 6.1s)

@tiagochst, any ideas on how to tighten these up? I'm going to modify the examples in `NameRegion()`, `CloseBySingleRegion()`, and `OrderCpGsByLocation()`. I don't remember writing any part of `AnnotateResults()`. Also, are the "checking examples" and "checking examples with --run-donttest" options duplicates of each other? We only have one function with \donttest{}.

**Comment by gabrielodom** [comment]
After @tiagochst's changes, the real time for the check (on Mac) is 13'39", and the computing time is 11'54".

I'll work on a few other things later tonight, and test it on my Windows machine in the office tomorrow.

**Comment by gabrielodom** [comment]
Timings after updates to the examples:

- creating vignettes (1m 28.2s)
- checking whether package ‘coMethDMR’ can be installed (38.7s)
- checking R code for possible problems (42.5s)
- checking for code/documentation mismatches (35.1s)
- checking examples (1m 9.9s)
    + CoMethAllRegions: 25.2
    + AnnotateResults: 8.3
- checking examples with --run-donttest (1m 28.5s)
- Running ‘testthat.R’ [31s/38s] (38.3s)
- checking for unstated dependencies in vignettes (38.5s)
- checking re-building of vignette outputs (2m 53.2s)

Total R elapsed time: 10'56"

When I run the example for `CoMethAllRegions()` in the console, it takes less than 2 seconds; no idea what's going on there.

**Comment by gabrielodom** [comment]
I discovered that R CMD check runs all the examples, including the components wrapped in \donttest{}. This had the additional side effect of creating the data set `closeByRegions.rds` even though we had wrapped the call to `WriteCloseByRegions()` in a \donttest{}. That was one of the CRAN notes. It's gone now that we just commented out the call in the example.

Also, I removed all calls to `CoMethAllRegions()` in the tests for `lmmTest()` and `lmmTestAllRegions()`, and manually replicated the output necessary to test those functions.
